RELEASE_NOTES.md

```diff
@@ -0,0 +1,9 @@
+Release Notes
+=============
+
+vX.X.X
```

Comment:
[Question] When you say 
> A release will give us a tag that can be used to link OpenAssetIO version to associated compatible OpenAssetIO-Test-CMake version.

do you mean that you don't know the appropriate tag number yet, and that's why this is X.X.X?

Reply:
Yeah from [our upstream docs](https://github.com/OpenAssetIO/OpenAssetIO/blob/main/contributing/CHANGES.md)

> whenever you add a change to the RELEASE_NOTES.md file as part of development, it should always be under a vX.X.X for the version at the top of the file. If you are the first to add a change, simply create this version heading as part of your commit.

Reply:
Right. Seems a bit academic if you're going to make a release pretty soon anyway, but nae problem.

Reply:
The motivation is that it makes it very clear to the 'next contributor' whether they need to start a new version section or not, or can amend/adapt existing release notes.